src/Entity/CivicrmEntity.php

```diff
@@ -17,13 +22,98 @@
  */
 class CivicrmEntity extends ContentEntityBase {
 
+  /**
+   * {@inheritdoc}
+   */
   public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
     $fields = [];
+    $civicrm_entity_info = SupportedEntities::getInfo()[$entity_type->id()];
+    $civicrm_required_fields = $entity_type->get('civicrm_required_fields');
```

Comment:
Isn't the `$entity_type->get('civicrm_required_fields')` derived from info from `SupportedEntities::getInfo()`? I wonder if we could just stash all the info from `SupportedEntities::getInfo()` on the entity type so we don't have to keep calling it? That would also mean that developers would have a place (ie. `hook_entity_type_alter()`) to alter this info if they wanted.

Not super important to this PR, just a thought :-)